suite/chaos/reorg_confirmer.go

```diff
@@ -0,0 +1,242 @@
+package chaos
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/chaos/experiments"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/environment"
+	"sort"
+	"sync"
+	"time"
+)
+
+const (
+	NextBlockTimeout    = 180 * time.Second
+	NetworkJoinAttempts = 30
+	NetworkJoinInterval = 5 * time.Second
+	VerifyAttempts      = 20
+	VerifyInterval      = 5 * time.Second
+)
+
+// NodeBlock block received from particular node
+type NodeBlock struct {
+	NodeID int
+	*types.Block
+}
+
+// ReorgConfirmer reorg stats collecting struct
+type ReorgConfirmer struct {
+	env                 environment.Environment
+	c                   client.BlockchainClient
+	joinBlockNumber     uint64
+	blockReceivers      []*AggregatingBlockReceiver
+	blocksByNodeMu      *sync.Mutex
+	reorgedBlocksByNode map[int][]*types.Block
+	blocksByNode        map[uint64]map[int]NodeBlock
+	blocksChan          chan NodeBlock
+}
+
+// NewReorgConfirmer creates a type that can create reorg chaos and confirm reorg has happened
+func NewReorgConfirmer(c client.BlockchainClient, env environment.Environment) (*ReorgConfirmer, error) {
```

Comment:
The `ReorgConfirmer` should be a header subscriber, not just use them. If you're using `retry` within function bodies, then something with the design is off.

Same with `awaitNetworkJoined`, you're awaiting an action based on new block headers, these sort of actions should be handled with subscribers, not retry functions.